aiohttp/cookiejar.py

```diff
@@ -194,15 +195,21 @@ def update_cookies(self,
 
         self._do_expiration()
 
-    def filter_cookies(self, request_url: URL=URL()) -> 'BaseCookie[str]':
+    def filter_cookies(self,
+                       request_url: URL=URL()
+                       ) -> Union['BaseCookie[str]', 'SimpleCookie[str]']:
         """Returns this jar's cookies filtered by their attributes."""
         self._do_expiration()
         if not isinstance(request_url, URL):
             warnings.warn("The method accepts yarl.URL instances only, got {}"
                           .format(type(request_url)),
                           DeprecationWarning)
             request_url = URL(request_url)
-        filtered = SimpleCookie()  # type: SimpleCookie[str]
+        filtered: Union[SimpleCookie[str], BaseCookie[str]]
+        if self._quote_cookie:
+            filtered = SimpleCookie()
+        else:
+            filtered = BaseCookie()
```

Comment:
I had in mind something like
```suggestion
        filtered = (  # type: Union[SimpleCookie[str], BaseCookie[str]]
            SimpleCookie() if self._quote_cookie
            else BaseCookie()
        )
```

Reply:
But what you wrote would work too.

Reply:
I like yours better too. I've commited this change.